test/contrib/lsf_test.py

```diff
@@ -103,5 +107,45 @@ def tearDown(self):
                 pass
 
 
+class TestSafeExtract(unittest.TestCase):
```

Comment:
Any test that is specific to only the SafeExtract should be contained within a test class (and file) specific to its source. In other words, these unit tests for SafeExtract belong in a `test/safe_extractor_test.py` file, not in a contrib location for a module which happens to use the SafeExtract functionality (unless you were wanting to test specific interactions between LSF and SafeExtract.

Reply:
correct, will fix it now